nlprule/src/rule/grammar.rs

```diff
@@ -58,13 +58,13 @@ impl Example {
     }
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct PosReplacer {
     pub(crate) matcher: PosMatcher,
 }
 
 impl PosReplacer {
-    fn apply(&self, text: &str, tokenizer: &Tokenizer) -> Option<String> {
+    pub fn apply(&self, text: &str, tokenizer: &Tokenizer) -> Vec<String> {
```

Comment:
A lot of API returns an iterator which could also just be an `impl Iterator` at the cost of a few lifetime parameters, in the long run this can make a large difference, since most `collect` calls could be avoided, which ends up both being an additive cost of N and an additional allocation (avoided).